src/App.tsx

```diff
@@ -1,61 +1,169 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import * as service from './api/service';
+
+import { Post } from './types/Post';
+import { Comment, CommentData } from './types/Comment';
+import { User } from './types/User';
+
 export const App: React.FC = () => {
+  const [userPosts, setUserPosts] = useState<Post[] | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [postComments, setPostComments] = useState<Comment[] | null>(null);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [isPostsError, setIsPostsError] = useState(false);
+  const [isCommentsError, setIsCommentsError] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isDisplayedForm, setIsDisplayedForm] = useState(false);
+
+  const loadUserPosts = (userId: number) => {
+    setUserPosts(null);
+    setIsLoadingPosts(true);
+    setIsPostsError(false);
+    service.getUserPosts(userId)
+      .then(posts => setUserPosts(posts))
+      .catch(() => setIsPostsError(true))
+      .finally(() => setIsLoadingPosts(false));
+  };
+
+  const loadPostComments = (postId: number) => {
```

Comment:
Do you really need to create everything in App.tsx?
I thinks it's better to split the logic into different components